<launch>
  <arg name="serial_no_camera1"    			default="750612070179"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera2"    			default="819612071257"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera3"    			default="817612071456"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera1"              			default="side_camera"/>		<!-- Note: Replace with camera name -->
  <arg name="camera2"              			default="knife_camera"/>		<!-- Note: Replace with camera name -->
  <arg name="camera3"              			default="tong_camera"/>		<!-- Note: Replace with camera name -->
  <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
  <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
  <arg name="tf_prefix_camera3"         default="$(arg camera3)"/>
  <arg name="initial_reset"             default="false"/>

  <group ns="$(arg camera1)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera1)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera1)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
    </include>
  </group>

  <group ns="$(arg camera2)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera2)"/>
      <arg name="tf_prefix"		          value="$(arg tf_prefix_camera2)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
    </include>
  </group>

  <group ns="$(arg camera3)" if="$(eval serial_no_camera3 != '')">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="serial_no"             value="$(arg serial_no_camera3)"/>
      <arg name="tf_prefix"		          value="$(arg tf_prefix_camera3)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
    </include>
  </group>
</launch>